### PR TITLE
First release cleanup

### PR DESCRIPTION
**Pull request by vbontempi**
### Description
First part of the cleanup task from https://github.com/terraform-ibm-modules/terraform-ibm-websphere-liberty-operator/issues/8

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

- Scaled down complete test cluster
- reviewed WebSphere Liberty naming
- Removed commented code
- Fixed kubeconfig path for datasource 
- Removed default region for extension
- Prefixed sample app URL with `https://` to make it clickable when deployed through DA
- Made resource groups names prefixed with random prefix with SLZ test override json

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [x] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,9 +13,9 @@ Update status and "latest release" badges:
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
 
 <!-- Add a description of module(s) in this repo -->
-Use this module to install a WebSphere® Liberty operator and create an instance of WebSphere Liberty operator on IBM Cloud Red Hat OpenShift cluster on VPC Gen2.
+Use this module to install WebSphere Application Server Liberty on IBM Cloud Red Hat OpenShift Container Platform on VPC landing zone by deploying IBM WebSphere Liberty operator.
```
Comment (rajatagarwal-ibm) [inline README.md:16]: Perhaps, `Use this module to install a WebSphere Application Server Liberty on the IBM Cloud Red Hat OpenShift Cluster`?

Comment (vbontempi) [reply README.md:16]: what about `Use this module to install WebSphere Application Server Liberty in IBM Cloud Red Hat OpenShift Container Platform on VPC landing zone: this module deploys and configures the IBM WebSphere Liberty operator into the IBM Cloud Red Hat OpenShift Cluster.`
?

@SirSpidey needing your help on this :-)

File: extensions/landing-zone/variables.tf [hunk 2]
```diff
@@ -7,7 +7,7 @@ variable "ibmcloud_api_key" {
 variable "region" {
   type        = string
   description = "Region to provision all resources created by this solution"
```
Comment (ocofaigh) [inline extensions/landing-zone/variables.tf:9]: Hmm, isn't this supposed to be the region of the cluster? Is there any way to pragmatically get that from cluster ID (maybe not if data lookup requires name)

Comment (vbontempi) [reply extensions/landing-zone/variables.tf:9]: Yes, fixing the description.
data lookup supports ID and so the region could be extracted from the cluster CRN through the datasource ibm_container_vpc_cluste, but unfortunately it needs the region for the ibm provider configuration

File: README.md (hunk 1 above)
Comment (SirSpidey) [reply README.md:16]: I think either of these 
> Use this module to install a WebSphere Liberty operator on your Red Hat OpenShift Container Platform on VPC landing zone.

> Use this module to install WebSphere Application Server Liberty on your Red Hat OpenShift Container Platform on VPC landing zone.

assuming that it is installing on an existing DA.

Comment (vbontempi) [reply README.md]: @SirSpidey thank you, adopted the first one

**Review by rajatagarwal-ibm — DISMISSED** [review]

File: extensions/landing-zone/kubeconfig/README.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+This directory must exist in source control so the `ibm_container_cluster_config` data lookup can use it to place the
```
Comment (SirSpidey) [inline extensions/landing-zone/kubeconfig/README.md:1]: Is this comment going to persist? And who sees it? We probably don't want to point to an internal issue from this module.

**Review by SirSpidey — CHANGES_REQUESTED** [review]
editorial updates (add periods to end to descriptions), and a couple questions.

**Review by SirSpidey — CHANGES_REQUESTED** [review]
editorial updates (add periods to end to descriptions), and a couple questions.

File: variables.tf [hunk 4]
```diff
@@ -30,26 +30,26 @@ variable "operator_helm_release_namespace" {
 
 variable "add_ibm_operator_catalog" {
   type        = bool
-  description = "Flag to configure the IBM Operator Catalog in the cluster before installing the WebSphere Liberty Operator. Default is true"
+  description = "Flag to configure the IBM Operator Catalog in the cluster before installing the IBM WebSphere Liberty operator. Default is true"
```
[the remark this answers is not in the document]
Comment (vbontempi) [reply variables.tf]: same for me.
Learned from https://www.ibm.com/docs/en/cloud-paks/1.0?topic=clusters-adding-operator-catalog

File: variables.tf [hunk 5]
```diff
@@ -30,26 +30,26 @@ variable "operator_helm_release_namespace" {
...
-  description = "Namespace to install the WebSphere Liberty Operator. Default to openshift-operators"
+  description = "Namespace to install the IBM WebSphere Liberty operator. Default to openshift-operators"
   default     = "openshift-operators"
   nullable    = false
 }
 
 variable "ws_liberty_operator_target_namespace" {
   type        = string
-  description = "Namespace to be watched by the WebSphere Liberty Operator. Default to null (operator to watch all namespaces)"
+  description = "Namespace to be watched by the IBM WebSphere Liberty operator. Default to null (operator to watch all namespaces)"
```
[the remark this answers is not in the document]
Comment (vbontempi) [reply variables.tf]: Watch should be known to operator adopters.
I don't think you can list a set of operators, 1 or none (which means all), according to doc

**Comment by vbontempi** [comment]
@SirSpidey I think I addressed all your comments and questions and pushed the changes

**Review by SirSpidey — DISMISSED** [review]
I'm fine with leaving the kubeconfig readme, I just don't understand what it does and why we would leave a link to an internal issue.

**Comment by vbontempi** [comment]
> I'm fine with leaving the kubeconfig readme, I just don't understand what it does and why we would leave a link to an internal issue.

right, the internal link is to be removed, done

**Comment by rajatagarwal-ibm** [comment]
Here `scripts/confirm-wsloperator-operational.sh`:
* `-o wide` flag doesn't work in the following command:
```
kubectl describe Subscription ibm-websphere-liberty -n "${namespace}" -o wide
```

**Comment by rajatagarwal-ibm** [comment]
Should update examples/complete/main.tf

* If we are not consuming 3 subnets, probably we can create only 1 subnet. https://github.com/terraform-ibm-modules/terraform-ibm-websphere-liberty-operator/blob/main/examples/complete/main.tf#L35
* If you agree to create 1 subnet, we should create only 1 public gateway as well. https://github.com/terraform-ibm-modules/terraform-ibm-websphere-liberty-operator/blob/main/examples/complete/main.tf#L26

**Comment by rajatagarwal-ibm** [comment]
The title is wrong in this readme extensions/landing-zone/README.md

**Comment by vbontempi** [comment]
> Need to update readme for the examples/complete/README.md.
> 
>     * Not creating 3 subnets anymore.
> 
>     * Not creating 3 worker pools anymore.

2 workers are the min for OCP. Reduced to 2.

All the other comments are addressed and pushed, please review

File: extensions/landing-zone/variables.tf [hunk 6]
```diff
@@ -1,48 +1,48 @@
...
   type        = string
-  description = "Namespace to install the WebSphere Liberty Operator. Default to openshift-operators"
+  description = "Namespace to install the IBM WebSphere Liberty operator. Default to openshift-operators."
   default     = "openshift-operators"
 }
 
 variable "ws_liberty_operator_target_namespace" {
   type        = string
-  description = "Namespace to be watched by the WebSphere Liberty Operator. Default to null (operator to watch all namespaces)"
+  description = "Namespace to be watched by the IBM WebSphere Liberty operator. Default to null (operator to watch all namespaces)."
```
Comment (SirSpidey) [inline extensions/landing-zone/variables.tf:45]:
```suggestion
  description = "Namespace that the the IBM WebSphere Liberty operator watches. Default is `null`, which means that the operator watches all the namespaces."
```

This didn't get updated from an earlier review.

**Review by SirSpidey — CHANGES_REQUESTED** [review]
a few more small suggestions

**Comment by vbontempi** [comment]
> a few more small suggestions

thank you for the suggestions, committed all of them, could you please review?

**Review by SirSpidey — APPROVED** [review]
also updated the settings.yml description in this PR.

**Comment by ocofaigh** [comment]
@vburckhardt @vbontempi @rajatagarwal-ibm The PR was not merged with a commit message that will generate a release. If you need a release like you indicated in the PR description, a new PR needs to be created and merged with correct prefix

**Pull request merged**